internal/resources/k6/data_source_k6_project_allowed_load_zones.go

```diff
@@ -0,0 +1,99 @@
+package k6
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/grafana/terraform-provider-grafana/v4/internal/common"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSourceWithConfigure = (*projectAllowedLoadZonesDataSource)(nil)
+)
+
+var (
+	dataSourceProjectAllowedLoadZonesName = "grafana_k6_project_allowed_load_zones"
+)
+
+func dataSourceProjectAllowedLoadZones() *common.DataSource {
+	return common.NewDataSource(
+		common.CategoryK6,
+		dataSourceProjectAllowedLoadZonesName,
+		&projectAllowedLoadZonesDataSource{},
+	)
+}
+
+// projectAllowedLoadZonesDataSourceModel maps the data source schema data.
+type projectAllowedLoadZonesDataSourceModel struct {
+	ProjectID        types.Int32 `tfsdk:"project_id"`
+	AllowedLoadZones types.List  `tfsdk:"allowed_load_zones"`
+}
+
+// projectAllowedLoadZonesDataSource is the data source implementation.
+type projectAllowedLoadZonesDataSource struct {
+	basePluginFrameworkDataSource
+}
+
+// Metadata returns the data source type name.
+func (d *projectAllowedLoadZonesDataSource) Metadata(_ context.Context, _ datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = dataSourceProjectAllowedLoadZonesName
+}
+
+// Schema defines the schema for the data source.
+func (d *projectAllowedLoadZonesDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Retrieves allowed load zones for a k6 project.",
+		Attributes: map[string]schema.Attribute{
+			"project_id": schema.Int32Attribute{
+				Description: "The identifier of the project to retrieve allowed load zones for.",
+				Required:    true,
+			},
+			"allowed_load_zones": schema.ListAttribute{
+				Description: "List of allowed k6 load zone IDs for this project.",
```

Comment:
```suggestion
			"allowed_load_zones": schema.ListAttribute{
				Description: "List of allowed k6 private load zones (names) for this project.",
```

I've wondered about the naming here: this object is supposed to be only for configuring private LZs, in reality. AFAIK, this is not going to change in the future either.
Changing the `allowed_load_zones` to `allowed_private_load_zones` seems a bit too verbose ATM, but perhaps, at least the description could contain clarification?
cc @pmdz @fornfrey

Reply:
I've tweaked the description a little bit in b2396268ceb79670aa9887fb1b3a5a3ead5c3369

This are not names btw., we're referencing the IDs here 
<img width="204" height="179" alt="image" src="https://github.com/user-attachments/assets/f19e4664-d3e9-4592-b35b-527695fc1b40" />

Reply:
Thanks, Pawel!
From user's perspective, this can be confusing actually, because it's a `name` in Kubernetes resource. But in UI, we call it "ID" :smile: But you're right that this is a frontend level problem: I've brought it up before in the Slack too. 
cc @dgzlopes for visibility: just in case, we should change this everywhere.